service/controller/clusterapi/v28/adapter/guest_auto_scaling_group.go

```diff
@@ -35,7 +35,7 @@ func (a *GuestAutoScalingGroupAdapter) Adapt(cfg Config) error {
 	}
 
 	{
-		numAZs := len(key.StatusAvailabilityZones(cfg.MachineDeployment))
+		numAZs := len(key.WorkerAvailabilityZones(cfg.MachineDeployment))
```

Comment:
`StatusAvailabilityZones()` now returns also an error which must be handled (and internally it does quite a bit of subnet computation) so simpler `WorkerAvailabilityZones()` was more convenient in places where only the worker AZ count was important. This is a bit ugly right now, but these will go away once we separate workers to node pools.